src/core/worker.js

```diff
@@ -542,24 +543,53 @@ class WorkerMessageHandler {
           pdfManager.ensureDoc("startXRef"),
           pdfManager.ensureDoc("xref"),
           pdfManager.ensureDoc("linearization"),
+          pdfManager.ensureCatalog("structTreeRoot"),
         ];
         const promises = [];
 
         const newAnnotationsByPage = !isPureXfa
           ? getNewAnnotationsMap(annotationStorage)
           : null;
-        const [stream, acroForm, acroFormRef, startXRef, xref, linearization] =
-          await Promise.all(globalPromises);
+        const [
+          stream,
+          acroForm,
+          acroFormRef,
+          startXRef,
+          xref,
+          linearization,
+          _structTreeRoot,
+        ] = await Promise.all(globalPromises);
+        let structTreeRoot;
```

Comment:
It seems that we're now looking up the raw "Root" a bunch of times in this code. Can we please do that *once* here instead:
```suggestion
        ] = await Promise.all(globalPromises);
        const rootRef = xref.trailer.getRaw("Root") || null;
        let structTreeRoot;
```
... and change all call-sites to use it:
 - https://github.com/mozilla/pdf.js/blob/469202c98a00aa779dbb015b75dfdf03040593ab/src/core/worker.js#L568
 - https://github.com/mozilla/pdf.js/blob/469202c98a00aa779dbb015b75dfdf03040593ab/src/core/worker.js#L610
 - https://github.com/mozilla/pdf.js/blob/469202c98a00aa779dbb015b75dfdf03040593ab/src/core/worker.js#L702